wpimath/src/main/java/edu/wpi/first/math/MathUtil.java

```diff
@@ -35,24 +35,57 @@ public static double clamp(double value, double low, double high) {
 
   /**
    * Returns 0.0 if the given value is within the specified range around zero. The remaining range
-   * between the deadband and 1.0 is scaled from 0.0 to 1.0.
+   * between the deadband and the maximum magnitude is scaled from 0.0 to the maximum magnitude (or
+   * left unchanged if the maximum magnitude is infinite).
    *
    * @param value Value to clip.
    * @param deadband Range around zero.
+   * @param maxMagnitude The maximum magnitude of the input.
    * @return The value after the deadband is applied.
    */
-  public static double applyDeadband(double value, double deadband) {
+  public static double applyDeadband(double value, double deadband, double maxMagnitude) {
     if (Math.abs(value) > deadband) {
+      if (maxMagnitude == Double.POSITIVE_INFINITY) {
+        return value;
+      }
       if (value > 0.0) {
-        return (value - deadband) / (1.0 - deadband);
+        // Map [0, deadband] to 0 and map [deadband, max] to [0, max].
+        //
+        // y - y₁ = m(x - x₁)
+        // y - y₁ = (y₂ - y₁)/(x₂ - x₁) (x - x₁)
+        // y = (y₂ - y₁)/(x₂ - x₁) (x - x₁) + y₁
+        //
+        // (x₁, y₁) = (deadband, 0) and (x₂, y₂) = (max, max).
+        // x₁ = deadband
+        // y₁ = 0
+        // x₂ = max
+        // y₂ = max
+        //
+        // y = (max - 0)/(max - deadband) (x - deadband) + 0
+        // y = max/(max - deadband) (x - deadband)
+        // y = max (x - deadband)/(max - deadband)
+        return maxMagnitude * (value - deadband) / (maxMagnitude - deadband);
```

Comment:
It seems like this will run into floating point issues with large values of maxMagnitude and small values of deadband?

Reply:
In those cases you should set the param to infinity because the algorithm is barely doing anything anyway